File: ios/quickstarts/digitalinkrecognition/DigitalInkRecognitionExampleObjC/RecognizedInk.m
#import "RecognizedInk.h"

#import <Foundation/Foundation.h>
#import <UIKit/UIKit.h>

@import MLKit;

NS_ASSUME_NONNULL_BEGIN

@implementation RecognizedInk : NSObject

- (nullable instancetype)initWithInk:(MLKInk *)ink {
  self = [super init];
  if (self != nil) {
    _ink = ink;
  }
  return self;
}

@end

NS_ASSUME_NONNULL_END

File: ios/quickstarts/digitalinkrecognition/DigitalInkRecognitionExample/RecognizedInk.swift
import Foundation

import MLKit

class RecognizedInk: NSObject {

  /** Ink, displayed to the user. */
  var ink: Ink
  /** Top recognition result candidate for the ink, displayed to the user. */
  var text: String? = nil

  /**
   * Creates a `RecognizedInk` with the given `Ink` once it is sent for recognition. Text is initially
   * empty.
   * @param ink The ink to be stored in the `RecognizedInk` object.
   * @return the initialized `RecognizedInk` object.
   */
  init(ink: Ink) {
    self.ink = ink
  }

}
